### PR TITLE
[AEA-1002] proper handling upgrade and remove dependencies for non vendor packages

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

proper handling upgrade and remove dependencies for non vendor packages

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/remove.py [hunk 1]
```diff
@@ -373,36 +386,9 @@ def agent_name(self) -> str:
...
-                    )
-                )
-            if (
-                item_folder.exists()
-                and not self.agent_config.author == self.item_id.author
-            ):  # pragma: no cover
-                raise click.ClickException(
-                    "{} {} author is different from {} agent author. "
-                    "Please fix the author field.".format(
-                        self.item_name, self.item_type, self.agent_name
```
Comment (DavidMinarsch) [inline aea/cli/remove.py:399]: This check should be maintained. It could be moved under `get_component_directory`?

**Review by DavidMinarsch — APPROVED** [review]
LGTM

File: aea/cli/remove.py (hunk 1 above)
Comment (solarw) [reply aea/cli/remove.py:399]: did in `get item config` but it's better to do an overall refactoring, cause component config loading and checks are done in various code places, so the codebase can be unified.

**Review by marcofavorito — APPROVED** [review]
LGTM

**Pull request merged**